File: push/src/main/java/com/xiaomi/tinyData/TinyDataCacheProcessor.java
package com.xiaomi.tinyData;

import android.content.Context;
import android.text.TextUtils;
import com.xiaomi.channel.commonutils.android.DeviceInfo;
import com.xiaomi.channel.commonutils.logger.MyLog;
import com.xiaomi.channel.commonutils.network.Network;
import com.xiaomi.push.service.DefaultConfig;
import com.xiaomi.push.service.MIPushAccountUtils;
import com.xiaomi.push.service.OnlineConfig;
import com.xiaomi.push.service.PushConstants;
import com.xiaomi.push.service.PushServiceConstants;
import com.xiaomi.push.service.TinyDataStorage;
import com.xiaomi.push.service.PingCallBack;
import com.xiaomi.push.service.XMPushService;
import com.xiaomi.xmpush.thrift.ConfigKey;
import java.io.File;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/tinyData/TinyDataCacheProcessor.class */
public class TinyDataCacheProcessor implements PingCallBack {
    private static final int DEFAULT_LAST_TINY_DATA_UPLOAD_TIMESTAMP = -1;
    public static final String LAST_TINY_DATA_UPLOAD_TIMESTAMP = "last_tiny_data_upload_timestamp";
    private static boolean mIsTinyDataExtracting = false;
    private Context mContext;
    private int mPeriod;
    private boolean mUploadSwitch;

    public TinyDataCacheProcessor(Context context) {
        this.mContext = context;
    }

    private boolean canUpload(TinyDataUploader tinyDataUploader) {
        if (!Network.hasNetwork(this.mContext) || tinyDataUploader == null || TextUtils.isEmpty(getAppId(this.mContext.getPackageName())) || !new File(this.mContext.getFilesDir(), TinyDataStorage.TINY_DATA_CACHE_FILE_NAME).exists() || mIsTinyDataExtracting) {
            return false;
        }
        return !OnlineConfig.getInstance(this.mContext).getBooleanValue(ConfigKey.ScreenOnOrChargingTinyDataUploadSwitch.getValue(), false) || DeviceInfo.isCharging(this.mContext) || DeviceInfo.isScreenOn(this.mContext);
    }

    private String getAppId(String str) {
        return PushConstants.PUSH_SERVICE_PACKAGE_NAME.equals(str) ? MIPushAccountUtils.MIPUSH_MIUI_APPID : this.mContext.getSharedPreferences(PushServiceConstants.PREF_KEY_REGISTERED_PKGS, 0).getString(str, null);
    }

    private void readOnlineConfig(Context context) {
        this.mUploadSwitch = OnlineConfig.getInstance(context).getBooleanValue(ConfigKey.TinyDataUploadSwitch.getValue(), true);
        int intValue = OnlineConfig.getInstance(context).getIntValue(ConfigKey.TinyDataUploadFrequency.getValue(), DefaultConfig.DEFAULT_TINY_DATA_UPLOAD_FREQUNECY);
        this.mPeriod = intValue;
        this.mPeriod = Math.max(60, intValue);
    }

    public static void setIsTinyDataExtracting(boolean z) {
        mIsTinyDataExtracting = z;
    }

    private boolean verifyUploadPeriod() {
        return Math.abs((System.currentTimeMillis() / 1000) - this.mContext.getSharedPreferences("mipush_extra", 4).getLong(LAST_TINY_DATA_UPLOAD_TIMESTAMP, -1L)) > ((long) this.mPeriod);
    }

    @Override // com.xiaomi.push.service.PingCallBack
    public void pingFollowUpAction() {
        readOnlineConfig(this.mContext);
        if (this.mUploadSwitch && verifyUploadPeriod()) {
            MyLog.w("TinyData TinyDataCacheProcessor.pingFollowUpAction ts:" + System.currentTimeMillis());
            TinyDataUploader uploader = TinyDataManager.getInstance(this.mContext).getUploader();
            if (canUpload(uploader)) {
                mIsTinyDataExtracting = true;
                TinyDataCacheReader.addTinyDataCacheReadJob(this.mContext, uploader);
            } else {
                MyLog.w("TinyData TinyDataCacheProcessor.pingFollowUpAction !canUpload(uploader) ts:" + System.currentTimeMillis());
            }
        }
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/tinyData/TinyDataCacheProcessor.kt
package com.xiaomi.tinyData

import android.content.Context
import android.text.TextUtils
import com.xiaomi.channel.commonutils.android.DeviceInfo
import com.xiaomi.channel.commonutils.logger.LoggerInterface
import com.xiaomi.channel.commonutils.logger.MyLog
import com.xiaomi.channel.commonutils.network.Network
import com.xiaomi.push.service.DefaultConfig
import com.xiaomi.push.service.MIPushAccountUtils
import com.xiaomi.push.service.OnlineConfig
import com.xiaomi.push.service.PushConstants
import com.xiaomi.push.service.PushServiceConstants
import com.xiaomi.push.service.TinyDataStorage
import com.xiaomi.push.service.PingCallBack
import java.io.File

class TinyDataCacheProcessor(private val mContext: Context) : PingCallBack {
    private var mPeriod: Int = 0
    private var mUploadSwitch: Boolean = false

    private fun canUpload(tinyDataUploader: TinyDataUploader?): Boolean {
        if (!Network.hasNetwork(mContext) ||
            tinyDataUploader == null ||
            TextUtils.isEmpty(getAppId(mContext.packageName)) ||
            !File(mContext.filesDir, TinyDataStorage.TINY_DATA_CACHE_FILE_NAME).exists() ||
            mIsTinyDataExtracting
        ) {
            return false
        }
        return !OnlineConfig.getInstance(mContext).getBooleanValue(
            com.xiaomi.xmpush.thrift.ConfigKey.ScreenOnOrChargingTinyDataUploadSwitch.value,
            false
        ) || DeviceInfo.isCharging(mContext) || DeviceInfo.isScreenOn(mContext)
    }

    private fun getAppId(str: String): String {
        return if (PushConstants.PUSH_SERVICE_PACKAGE_NAME == str) {
            MIPushAccountUtils.MIPUSH_MIUI_APPID
        } else {
            mContext.getSharedPreferences(PushServiceConstants.PREF_KEY_REGISTERED_PKGS, 0).getString(str, null)
        } ?: ""
    }

    private fun readOnlineConfig(context: Context) {
        mUploadSwitch = OnlineConfig.getInstance(context).getBooleanValue(
            com.xiaomi.xmpush.thrift.ConfigKey.TinyDataUploadSwitch.value,
            true
        )
        val intValue = OnlineConfig.getInstance(context).getIntValue(
            com.xiaomi.xmpush.thrift.ConfigKey.TinyDataUploadFrequency.value,
            DefaultConfig.DEFAULT_TINY_DATA_UPLOAD_FREQUNECY
        )
        mPeriod = maxOf(60, intValue)
    }

    private fun verifyUploadPeriod(): Boolean {
        val lastUploadTime = mContext.getSharedPreferences("mipush_extra", 4)
            .getLong(LAST_TINY_DATA_UPLOAD_TIMESTAMP, DEFAULT_LAST_TINY_DATA_UPLOAD_TIMESTAMP.toLong())
        return kotlin.math.abs((System.currentTimeMillis() / 1000) - lastUploadTime) > mPeriod.toLong()
    }

    override fun pingFollowUpAction() {
        readOnlineConfig(mContext)
        if (mUploadSwitch && verifyUploadPeriod()) {
            MyLog.w("TinyData TinyDataCacheProcessor.pingFollowUpAction ts:${System.currentTimeMillis()}")
            val uploader = TinyDataManager.getInstance(mContext)?.uploader
            if (canUpload(uploader)) {
                mIsTinyDataExtracting = true
                TinyDataCacheReader.addTinyDataCacheReadJob(mContext, uploader)
            } else {
                MyLog.w("TinyData TinyDataCacheProcessor.pingFollowUpAction !canUpload(uploader) ts:${System.currentTimeMillis()}")
            }
        }
    }

    companion object {
        private const val DEFAULT_LAST_TINY_DATA_UPLOAD_TIMESTAMP = -1
        const val LAST_TINY_DATA_UPLOAD_TIMESTAMP = "last_tiny_data_upload_timestamp"
        private var mIsTinyDataExtracting = false

        fun setIsTinyDataExtracting(isExtracting: Boolean) {
            mIsTinyDataExtracting = isExtracting
        }
    }
}
